sim-lib/src/lib.rs

```diff
@@ -220,6 +254,7 @@ impl Simulation {
             if let Err(e) = res {
                 log::error!("Task exited with error: {e}");
                 success = false;
+                shutdown_trigger.trigger();
```

Comment:
Why do we need to have the trigger here? If one of the threads exits (which means we're done awaiting) then we should have exited?

Reply:
Tested without this in here and still works ✅